controllers/cloud.redhat.com/providers/autoscaler/simple.go

```diff
@@ -0,0 +1,148 @@
+package autoscaler
+
+import (
+	"fmt"
+	"log"
+
+	res "k8s.io/apimachinery/pkg/api/resource"
+
+	crd "github.com/RedHatInsights/clowder/apis/cloud.redhat.com/v1alpha1"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/config"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers"
+	deployProvider "github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers/deployment"
+	apps "k8s.io/api/apps/v1"
+	v2 "k8s.io/api/autoscaling/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ProvideSimpleAutoScaler(app *crd.ClowdApp, appConfig *config.AppConfig, sp *providers.Provider, deployment crd.Deployment) error {
+	cachedDeployment, err := getDeploymentFromCache(&deployment, app, sp)
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+	deploymentHPA := makeDeployemntSimpleHPA(&deployment, app, appConfig, cachedDeployment)
+	hpaResource := deploymentHPA.getResource()
+
+	err = sp.Client.Create(sp.Ctx, &hpaResource)
+	if err != nil {
+		fmt.Println("HPA Error: ", err)
+	}
+	return nil
+}
+
+//Get the core apps.Deployment from the provider cache
+//This is in AutoScalerSimpleProvider because we need access to the provider cache
+func getDeploymentFromCache(clowdDeployment *crd.Deployment, app *crd.ClowdApp, sp *providers.Provider) (*apps.Deployment, error) {
```

Comment:
Throughout the code, when we `get/generate` a k8s resource for the first time, we tend to follow the k8s pattern of population, rather than returning it. Throughout the Clowder code base, particularly when using the cache and clients you will see that we create an empty object, and send a pointer for population, rather than expect an object to be returned to us.